frontend/src/component/pages/Home.tsx

```diff
@@ -91,6 +93,17 @@ export class Home extends Component<Props, State> {
     });
     this.handleStateChange();
     this.autoFillLongLink();
+
+    const changeLog = await this.props.updatesService.getChangeLog();
+    this.setState({ changeLog });
+    const lastSeenTimestamp = await this.props.updatesService.getLastSeenChangeLog();
+    if (
+      changeLog &&
+      changeLog[0] &&
+      lastSeenTimestamp < changeLog[0].releasedAt
+    ) {
+      this.handleShowChangeLogBtnClick();
+    }
```

Comment:
```suggestion
    if (!changeLog) {
        return;
    }
    
    if (changeLog.length < 1) {
        return;
    }
    const lastSeenAt = await this.props.updatesService.getLastSeenAt();
   if (lastSeenAt >= changeLog[0].releasedAt) {
       return;
    )
   this.showChangeLogs();
```

Reply:
Instead of doing this, let's encapsulate the logic inside `ChangeLogService` to reduce the complexity of `Home` page.:

```typescript
if(!this.props.updatesService.hasUpdates()) {
    return false;
}
// setState is async
 this.setState({ changeLog }, ()=>{
      const changeLog = await this.props.updatesService.getChangeLog();
 });
this.showChangeLogs();
```

Reply:
The we don't need to expose `getLastSeenChangeLog`